openstack/utils/templates/_ini_sections.tpl

```diff
@@ -66,3 +66,14 @@ logging_default_format_string = %(asctime)s %(process)d %(levelname)s %(name)s [
 logging_user_identity_format = %(user)s %(tenant)s %(domain)s %(user_domain)s %(project_domain)s
 logging_exception_prefix = %(asctime)s %(process)d ERROR %(name)s %(resource)s%(instance)s
 {{- end }}
+
+{{- define "ini_sections.coordination" -}}
+[coordination]
+backend_url = {{ if eq .Values.coordinationBackend "memcached" -}}
+    memcached://{{ .Chart.Name }}-memcached.{{ include "svc_fqdn" . }}:{{ .Values.memcached.memcached.port | default 11211 }}
+{{- else if eq .Values.coordinationBackend "file" -}}
+    file://$state_path/coordination
```

Comment:
I can do that, I am not insisting on this approach, but let me explain the reasoning behind it:
From my perspective, you are viewing it from the chart down, while my choice is "bottom-up":

We have four (or five) components, which need to play together:
0. The application defaults in code: There a $state_path and the name suggest that all state should be stored underneath it, and is by default `/var/lib/$SERVICE`
1. The Dockerfile, which creates the filesystem layout and also that directory, so that the default setting works
2. The configuration file, which allows us to change `$state_path` as well as the coordination path in `backend_url`
3a. Helm-Chart: The Deployment-spec, which allows us to specify the path to mount the directory, and to my knowledge requires that the parent directory exists.
4. The name of the chart, which is the same as the service.

If any of those things change, things will likely break. But my understanding of kubernetes and how we use the chart, that is okay, and in this case even the goal.

We could define for example simply a `/run/{{.Chart.name}}-coordination` (or even just `/run/coordination`) directory, and all the dependencies would go away.

But personally, I want to stay at close as possible at the openstack defaults, so someone knowning openstack knows their way around, and I consider it a plus, that things will break, if standards/defaults are broken.

Reply:
Added a comment